zookeeper/source/zk_dn_client.cc

```diff
@@ -84,64 +84,81 @@ namespace zkclient{
 
 		LOG(INFO) << "About to create the znode for the data ip";
 		if (zk->exists(BLOCK_LOCATIONS + std::to_string(uuid), exists, error_code)) {
-        	if (exists) {
-        		// TODO: Add to spec the location to which we're writing the size
-        		// zk->set(BLOCK_LOCATIONS + std::to_string(uuid), ZKWrapper::EMPTY_VECTOR, error_code); 
+			if (exists) {
+				// TODO: Add to spec the location to which we're writing the size
+				// zk->set(BLOCK_LOCATIONS + std::to_string(uuid), ZKWrapper::EMPTY_VECTOR, error_code);
 				// Add this datanode as the block's location in block_locations
 				DataNodeZNode znode_data;
 				strcpy(znode_data.ipPort, id.c_str());
 				std::vector<std::uint8_t> data(sizeof(znode_data));
```

Comment:
I'm not sure why it's written to the ZNode when it's the same as the end of the path. It looks like @Swyrough  added it [here](https://github.com/Rice-Comp413-2016/RDFS/commit/6944bf6019e6c778fbbb99cf5a4ab7eb38f26b05), but then @captain695 changed it [here](https://github.com/Rice-Comp413-2016/RDFS/commit/8fcae55022658ec40da2a054b2c646a4ec48f806) so the empty vector is written instead of the data.
If that's correct, we should also remove the part that creates the data, because otherwise it looks like the empty vector being written is a mistake.